runtime/src/main/java/org/corfudb/runtime/object/VersionLockedObject.java

```diff
@@ -672,25 +678,34 @@ protected void syncStreamUnsafe(ISMRStream stream, long timestamp) {
         log.trace("Sync[{}] {}", this, (timestamp == Address.OPTIMISTIC)
                 ? "Optimistic" : "to " + timestamp);
         long syncTo = (timestamp == Address.OPTIMISTIC) ? Address.MAX : timestamp;
-        stream.streamUpTo(syncTo)
-                .forEachOrdered(entry -> {
-                    try {
-                        Object res = applyUpdateUnsafe(entry, timestamp);
-                        if (timestamp == Address.OPTIMISTIC) {
+
+        Histogram histogram = VloMetricsHelper.metrics.histogram(syncStreamCount);
+        final long[] counter = new long[1];
```

Comment:
why we need an array here? 
It seems that we need AtomicLong, or (I prefer) just use a plain old for cycle

Reply:
If I use a plain for, I have to do `stream.streamUpTo(syncTo).collect(Collectors.toList())`.
Will it bring extra copy?

Reply:
yep, looks like, so please take a look at my comment about the counter, would be nice if you agree

Reply:
Yeah, I am using it right now.